src/Kestrel.Core/Internal/HttpsConnectionMiddleware.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipelines;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Https.Internal
+{
+    public class HttpsConnectionMiddleware
```

Comment:
hi @davidfowl can I use this middleware to get tls support for the connection abstraction based mqtt handler I created?
(does this work with tcp or is http(s) required)

https://github.com/chkr1011/MQTTnet/blob/master/Source/MQTTnet.AspnetCore/MqttConnectionHandler.cs

Reply:
Yes the idea is to have a TlsMiddleware that's separate from Kestrel that can be placed in front of any `ConnectionHandler`. This is the first go to do that within Kestrel itself to see what issues exist.